include/drivers/adp536x.h

```diff
@@ -4,19 +4,182 @@
+#define ADP536X_OC_CHRG_THRESHOLD_50mA  0x01
+#define ADP536X_OC_CHRG_THRESHOLD_100mA 0x02
+#define ADP536X_OC_CHRG_THRESHOLD_150mA 0x03
+#define ADP536X_OC_CHRG_THRESHOLD_200mA 0x04
+#define ADP536X_OC_CHRG_THRESHOLD_250mA 0x05
+#define ADP536X_OC_CHRG_THRESHOLD_300mA 0x06
+#define ADP536X_OC_CHRG_THRESHOLD_400mA 0x07
+
+/**
+ * @brief Initialize ADP536X.
+ *
+ * Retrieves the device structure of the ADP536X device.
+ *
+ * @param[in] dev_name Pointer to the device name.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_init(const char *dev_name);
+
+/**
+ * @brief Set the VBUS current limit.
+ *
+ * @param[in] value The upper current threshold in LSB.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_vbus_current_set(u8_t value);
+
+/**
+ * @brief Set the charger current.
+ *
+ * @param[in] value The charger current in LSB.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_charger_current_set(u8_t value);
+
+/**
+ * @brief Set the charger termination voltage.
+ *
+ * Configures the maximum battery voltage where the charger
+ * will remain active.
+ *
+ * @param[in] value The charger termination voltage.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_charger_termination_voltage_set(u8_t value);
+
+/**
+ * @brief Enable charging.
+ *
+ * @param[in] enable The desired charger operation state.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_charging_enable(bool enable);
+
+/**
+ * @brief Read the STATUS1 register.
+ *
+ * @param[out] buf The read value of the STATUS1 register.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_charger_status_1_read(u8_t *buf);
+
+/**
+ * @brief Read the STATUS2 register.
+ *
+ * @param[out] buf The read value of the STATUS2 register.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_charger_status_2_read(u8_t *buf);
+
+/**
+ * @brief Enable charge hiccup protection mode.
+ *
+ * @param[in] enable The desired hiccup operation state.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_oc_chg_hiccup_set(bool enable);
+
+/**
+ * @brief Enable discharge hiccup protection mode.
+ *
+ * @param[in] enable The desired hiccup operation state.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_oc_dis_hiccup_set(bool enable);
+
+/**
+ * @brief Enable the buck and boost regulators.
```

Comment:
The regulator is a combined buck-boost converter, so it's actually only one regulator affected by this function

Reply:
Noted!